tests/api_test.py

```diff
@@ -1919,6 +1919,20 @@ def f(x1, x2, g):
     ):
       jax.vmap(f, (0, 0, None))(jnp.ones(2), jnp.ones(3), jnp.add)
 
+  def test_vmap_inconsistent_sizes_constructs_proper_error_message_kwargs(self):
+    def f(x1, x2, a3):
+      return jnp.add(x1, x2, a3)
+
+    with self.assertRaisesRegex(
+      ValueError,
+      "vmap got inconsistent sizes for array axes to be mapped:\n"
+      r"  \* most axes \(2 of them\) had size 2, e.g. axis 0 of argument x1 of type float32\[2\];\n"
+      r"  \* one axis had size 1: axis 0 of kwargs\['a3'\] of type float32\[1\]",
+    ):
+      # previously this test would fail when kwargs were passed in a different order
+      # that what's in the function signature: issue #24406
```

Comment:
You can remove this comment I think: the "regression test" comment above is sufficient